momento-sdk/src/main/java/momento/sdk/config/transport/storage/StorageGrpcConfiguration.java

```diff
@@ -0,0 +1,260 @@
+package momento.sdk.config.transport.storage;
+
+import static momento.sdk.ValidationUtils.ensureRequestDeadlineValid;
+
+import java.time.Duration;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import momento.sdk.internal.GrpcChannelOptions;
+
+/** Abstracts away the gRPC configuration tunables. */
+public class StorageGrpcConfiguration {
+
+  private final Duration deadline;
+  private final int minNumGrpcChannels;
+  private final Optional<Integer> maxMessageSize;
+  private final Optional<Boolean> keepAliveWithoutCalls;
+  private final Optional<Integer> keepAliveTimeoutMs;
+  private final Optional<Integer> keepAliveTimeMs;
+
+  /**
+   * Constructs a StorageGrpcConfiguration.
+   *
+   * @param deadline The maximum duration of a gRPC call.
+   */
+  public StorageGrpcConfiguration(@Nonnull Duration deadline) {
+    this(
+        deadline,
+        1,
+        Optional.of(GrpcChannelOptions.DEFAULT_MAX_MESSAGE_SIZE),
+        Optional.of(GrpcChannelOptions.DEFAULT_KEEPALIVE_WITHOUT_STREAM),
+        Optional.of(GrpcChannelOptions.DEFAULT_KEEPALIVE_TIMEOUT_MS),
+        Optional.of(GrpcChannelOptions.DEFAULT_KEEPALIVE_TIME_MS));
+  }
+
+  /**
+   * Constructs a StorageGrpcConfiguration.
+   *
+   * @param deadline The maximum duration of a gRPC call.
+   * @param minNumGrpcChannels The minimum number of gRPC channels to keep open at any given time.
+   * @param maxMessageSize The maximum size of a message (in bytes) that can be received by the
+   *     client.
+   * @param keepAliveWithoutCalls Whether to send keepalive pings without any active calls.
+   * @param keepAliveTimeout The time to wait for a keepalive ping response before considering the
+   *     connection dead.
+   * @param keepAliveTime The time to wait between keepalive pings.
+   */
+  public StorageGrpcConfiguration(
+      @Nonnull Duration deadline,
+      int minNumGrpcChannels,
```

Comment:
We would like to keep just the duration for MVP? ([JS-SDK reference](https://github.com/momentohq/client-sdk-javascript/pull/1318#discussion_r1646698476))

Reply:
The other settings are relevant for Lambda environments. That is usually in scope for MVP.

Reply:
Thanks for confirming. The PR looks good to me!